File: day01/day01.js
const input = require('./input.json');
const data = input.data;

const exerciseOne = () => {
    let measurements = data;
    let increases = 0;
    measurements.forEach((item, index) => {
        if (item > measurements[index - 1]) {
            increases++
        }
    });
    console.log(increases);
};

const exerciseTwo = () => {
    let measurements = data;
    let increases = 0;
    let windows = [];
    measurements.forEach((item, index) => {
        let secondValue = measurements[index + 1] || 0;
        let thirdValue = measurements[index + 2] || 0;
        if (secondValue > 0 && thirdValue > 0) {
            let result = item + secondValue + thirdValue;
            windows.push(result);
        }
    });
    windows.forEach((item, index) => {
        if (item < windows[index + 1]) {
            increases++;
        }
    });
    console.log(increases);
}

exerciseOne();
exerciseTwo();

File: python/day01/day01.py
f = open('./input.txt', 'r')
data = []
for x in f:
    data.append(int(x))

# Exercise 1
increases = 0
for i in range(len(data)):
    if (data[i] > data[i - 1]):
        increases += 1
print(increases)

# Exercise 2
increases = 0
prevResult = float('inf')
for i in range(len(data)-2):
    result = data[i] + data[i+1] + data[i+2]
    if (result > prevResult):
        increases += 1
    prevResult = result
print(increases)